lib/presentation/index.dart

```diff
@@ -56,11 +55,16 @@ class _IndexPage extends StatelessWidget {
             // clear navigation stack before going to new route
             routeNavigatorState.popUntil((route) => route.isFirst);
           }
-
-          indexedPages[state.tab.index]
-              .navigatorKey
-              .currentState
-              .pushNamed(state.route);
+          if (state.route == '/bible_series') {
```

Comment:
I don't think we should be doing this kind of logic in here. What would be better is passing all the arguments as an object to the state and including that as is in the pushNamed. We should also default it to an empty object in the event.